### PR TITLE
Bundle fixes

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
There are 2 fixes to the bundling routine here:
- bundles can be minified in parallel, no need to do it sequentially 
- minified version of the router bundle was missing

@IgorMinar could you please have a look?

//cc: @btford

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1067,25 +1067,17 @@ gulp.task('!bundle.js.prod', ['build.js.prod'], function() {
...
-          }
-        );
-      });
+  var q = require('q');
+  var minBundleConfig = {
+    sourceMaps: true,
+    minify: true
+  };
+
+  return q.all([
```
Comment (IgorMinar) [inline gulpfile.js]: does this actually make a difference? unless the bundler runs in a different process (which I don't believe is the case) then the whole task is cpu/thread- bound and parallelizing doesn't make things any faster.

File: gulpfile.js [hunk 2]
```diff
@@ -1250,7 +1242,6 @@ gulp.task('bundles.js', [
   '!test.bundle.js.dev']);
 
 gulp.task('build.js', ['build.js.dev', 'build.js.prod', 'build.js.cjs', 'bundles.js', 'benchpress.bundle']);
-
```
Comment (IgorMinar) [inline gulpfile.js]: please don't remove ws :)

**Comment by IgorMinar** [comment]
if this actually improves things then LGTM, please clean up and merge, but do measure first.

File: gulpfile.js (hunk 1 above)
Comment (pkozlowski-opensource) [inline gulpfile.js]: @IgorMinar oh, honestly for me eventual perf improvement was just a bonus as compared to the code improvement. I've decided to change this when adding config for a router and parallel version felt so much better / easier to write and read. 

So for me it was just to shorten the code and explicitly express that we don't need to run those things in sequence as the previous code implied.

**Comment by pkozlowski-opensource** [comment]
OK, will clean up and merge as it reads better. If we've got any perf gain here that would be a bonus, but I wasn't going after perf.

**Comment by IgorMinar** [comment]
sgtm

**Pull request closed without merging**